seedDatabase/seed.js

```diff
@@ -0,0 +1,209 @@
+      'element_length': obj['element_length'],
+      'sequence': obj.sequence
+    }, elementFilePath);
+
+  }
+
+  return [length, lectureCount, quizCount, exerciseCount, articleCount];
+
+};
+
+const seedSection = async (courseId) => {
+
+  let n = numOfSubs([4, 5, 6, 7, 8, 9, 10]);
+
+  let records = [];
+  let length = 0;
+  let lectureCount = 0;
+  let exerciseCount = 0;
+  let articleCount = 0;
+  let quizCount = 0;
+
+  for (let i = 0; i < n; i++) {
+    let obj = {};
+    obj['id'] = uuidv4();
+    obj['course_id'] = courseId;
+    obj['title'] = faker.git.commitMessage();
+    [obj['section_length'], obj.lectures, obj.quizzes, obj.exercises, obj.articles] = await seedElement(courseId, obj['id']);
+    obj['sequence'] = i + 1;
+
+    length += obj['section_length'];
+    lectureCount += obj.quizzes;
+    exerciseCount += obj.exercises;
+    articleCount += obj.articles;
+    quizCount += obj.quizzes;
+
+    obj['section_length'] = new Date(obj['section_length']);
+    obj['section_length'] = obj['section_length'].toDateString();
+
+    await writeRecord({
+      'id': obj['id'],
+      'course_id': obj['course_id'],
+      'title': obj['title'],
+      'section_length': obj['section_length'],
+      'lectures': obj.lectures,
+      'quizzes': obj.quizzes,
+      'exercises': obj.exercises,
+      'articles': obj.articles,
+      'sequence': obj.sequence
+    }, sectionFilePath);
+
+  }
+
+  return [length, n, lectureCount, exerciseCount, articleCount, quizCount];
+
+};
+
+const seedCourse = async(numOfCourses) => {
+
+  for (let i = 0; i < numOfCourses; i++) {
+    let obj = {};
+    obj['id'] = uuidv4();
+    [
+      obj['course_length'],
+      obj['total_sections'],
+      obj['total_lectures'],
+      obj['total_exercises'],
+      obj['total_articles'],
+      obj['total_quizzes']
+    ] = await seedSection(obj['id']);
+    obj['updated_at'] = new Date();
+    obj['updated_at'] = obj['updated_at'].toDateString();
+    obj['course_length'] = new Date(obj['course_length']);
+    obj['course_length'] = obj['course_length'].toDateString();
+
+    await writeRecord({
+      'id': obj.id,
+      'course_length': obj['course_length'],
+      'total_sections': obj['total_sections'],
+      'total_lectures': obj['total_lectures'],
+      'total_exercises': obj['total_exercises'],
+      'total_articles': obj['total_articles'],
+      'total_quizzes': obj['total_quizzes'],
+      'updated_at': obj['updated_at']
+    }, courseFilePath);
+  }
+  console.log('DONE');
+  return;
+};
+
+// SET 2500 AS BATCH SIZE WITHOUT BURNING OUT RAM
+const batch = 2500;
+
+// CAN WRITE SINGLE FILE FOR EACH TABLE BY CHANGING totalRuns = 4000
+// DEPENDING ON THE PREFERRED SIZE OF FILES, totalRuns CAN BE CUSTOMIZED
+const timer = new TaskTimer(1000);
+
+timer.add({
+  id: 'Task',
+  tickInterval: 10,
+  totalRuns: 1000,
+  callback(task) {
+    seedCourse(batch);
+    console.log(`${task.id} has run ${task.currentRuns} times.`);
+  }
+});
+
+timer.start();
```

Comment:
This is clever. I kind of pain in the butt that you have to do it, but it's nicely executed!